src/components/pages/Sources/SourcesList/AddSourceModal.tsx

```diff
@@ -23,37 +23,47 @@ import {DashboardContext} from '@contexts';
 
 import {AddSourceModalContainer} from './SourcesList.styled';
 
-const AddSourceModal = () => {
+type AddSourceFormValues = {
+  name: string;
+  uri: string;
+  token?: string;
+  username?: string;
+};
+
+const AddSourceModal: React.FC = () => {
+  const [form] = Form.useForm<AddSourceFormValues>();
   const {navigate} = useContext(DashboardContext);
 
   const [createSource, {isLoading}] = useCreateSourceMutation();
 
   const namespace = useAppSelector(selectNamespace);
 
-  const onFinish = (values: any) => {
+  const onFinish = (values: AddSourceFormValues) => {
     const {name, uri, token, username} = values;
 
-    const body = {
+    const body: SourceWithRepository = {
       name,
       repository: {
-        type: 'git' as RepositoryTypeEnum,
+        type: 'git',
         uri,
         ...(username ? {usernameSecret: {name: username}} : {}),
         ...(token ? {tokenSecret: {name: token}} : {}),
       },
       namespace,
     };
 
-    createSource(body).then((res: any) => {
-      displayDefaultNotificationFlow(res, () => {
-        navigate(`/sources/${res.data.metadata.name}`);
+    createSource(body)
+      .unwrap()
```

Comment:
Why does it has to be unwrapped?

Reply:
It's the way RTK suggests to work with `res` in `then`.

Without `unwrap()`:
<img width="1154" alt="image" src="https://user-images.githubusercontent.com/25991946/234849789-6bf34f19-464b-48c2-bdca-91fddbcab9b8.png">

With `unwrap()`:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/25991946/234849907-4a7a5449-ac89-4855-98e3-2d1208d2eafa.png">